h2/src/main/org/h2/mvstore/DataUtils.java

```diff
@@ -1005,7 +1005,7 @@ public static int parseHexInt(String x) {
      * @return the parsed value
      * @throws IllegalStateException if parsing fails
      */
-    public static int readHexInt(HashMap<String, ?> map, String key, int defaultValue) {
+    public static int readHexInt(Map<String,?> map, String key, int defaultValue) {
```

Comment:
I the method ever used without HashMap? I understand Map is more generic, but the change seems unrelated to the issue. To make the patch smaller, I would avoid any unnecessary changes here.